presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java

```diff
@@ -618,6 +618,9 @@ public void checkCanRevokeRoles(ConnectorTransactionHandle transactionHandle, Co
     public void checkCanSetRole(ConnectorTransactionHandle transaction, ConnectorIdentity identity, AccessControlContext context, String role, String catalogName)
     {
         SemiTransactionalHiveMetastore metastore = getMetastore(transaction);
```

Comment:
Can you help me understand why we need to check for null as part of this bug fix?

Reply:
When a statement in a non-autocommit transaction fails, `InMemoryTransactionManager.abortInternal()` is triggered, which causes the relevant hive connector transaction manager to clean up it's corresponding connector transaction. However, the transaction in `InMemoryTransactionManager` is just marked as failed, but not removed, referring to https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java#L918.

Therefore, if we do not check the null value here, an NPE exception will be encountered when executing the rollback statement.